src/components/Animation/CreateAnimation.vue

```diff
@@ -87,81 +69,56 @@ export default {
-    this.$root.$off("localeChange", this.getModelRuns);
-    this.$root.$off("modelRunChanged", this.getModelRuns);
     this.$root.$off("redoAnimation", this.createMP4Handler);
     this.$root.$off("restoreState", this.restoreState);
-    this.$root.$off("timeLayerAdded", this.getModelRuns);
-    this.$root.$off("timeLayerRemoved", this.getModelRuns);
   },
   mixins: [datetimeManipulations],
-  props: ["map"],
   methods: {
     cancelAnimationCreation() {
       this.generating = false;
+      if (this.mapController) {
+        this.mapController.abort("");
+      }
+      if (this.animationController) {
+        this.animationController.abort("");
+      }
+      if (this.layersController) {
+        this.layersController.abort("");
+      }
     },
     cancelAnimationFromResize() {
       if (this.isAnimating) {
-        this.notifyCancelAnimateResize = true;
+        this.$root.$emit("cancelAnimationResize");
         this.cancelAnimationCreation();
       }
     },
-    getExtentHandler() {
-      this.getExtent();
-      this.$store.dispatch("Layers/setOutputWH", [
-        this.mapWidth,
-        this.mapHeight,
-      ]);
-    },
-    getExtent() {
-      const extent = this.map.getView().calculateExtent();
-      this.$store.dispatch("Layers/setExtent", extent);
-    },
-    getTimeTitleWidths(mapDivWidth) {
-      if (mapDivWidth < 600) {
-        return [280, 300];
-      } else {
-        return [mapDivWidth - 320, 300];
-      }
+    getTimeTitleWidths() {
+      return [this.mapWidth - 320, 300];
     },
     getModelRuns() {
-      let modelRuns = {};
-      const timeLayers = this.$mapLayers.arr.filter((l) =>
-        l.get("layerIsTemporal")
-      );
-      for (let i = 0; i < timeLayers.length; i++) {
-        if (timeLayers[i].get("layerCurrentMR") !== null) {
-          const refTime = timeLayers[i].get("layerCurrentMR");
-          if (refTime in modelRuns) {
-            modelRuns[refTime].push(timeLayers[i].get("layerName"));
-          } else {
-            modelRuns[refTime] = [timeLayers[i].get("layerName")];
-          }
-        }
-      }
-      if (Object.keys(modelRuns).length === 0) {
-        this.modelRunMessage = null;
-      } else if (Object.keys(modelRuns).length === 1) {
-        this.modelRunMessage = [
-          `${this.$t("ModelRun")}${this.$t("Colon")} ${this.localeDateFormat(
-            new Date(Date.parse(Object.keys(modelRuns)[0]))
-          )}`,
-        ];
-      } else {
-        let message = [];
-        for (const [key, values] of Object.entries(modelRuns)) {
-          message.push(
-            `${this.$t("ModelRun")}${this.$t("Colon")} ${this.localeDateFormat(
-              new Date(Date.parse(key))
-            )}`
+      let modelRuns = [];
+      let visibleLayers = this.$mapLayers.arr
+        .filter((l) => {
+          return l.get("layerVisibilityOn") && l instanceof OLImage;
+        })
+        .reverse();
+      for (let i = 0; i < visibleLayers.length; i++) {
```

Comment:
I recommend to change `visibleLayers.length` to a const variable to avoid having the for loop recalculate the `.length` each loop, since it's going to be the same.

```suggestion
      const numVisibleLayers = visibleLayers.length
      for (let i = 0; i < numVisibleLayers; i++) {
```